File: frontend/src/services/peopleservice.js
import restAPI from './api'
import UserService from './userservice'

export const search = (keyword) => restAPI.get(`/people/search/${keyword}`)

export const getContacts = () => {
  const user = UserService.getAuthenticatedUser()
  return restAPI.get(`/people/${user.id}/contacts`)
}

export const addContact = (contact) => {
  const user = UserService.getAuthenticatedUser()
  return restAPI.post(`/people/${user.id}/contacts/add`, contact)
}

export const removeContact = (contact) => {
  const user = UserService.getAuthenticatedUser()
  return restAPI.delete(`/people/${user.id}/contacts/remove`, contact)
}

const PeopleService = {
  search,
  getContacts,
  addContact,
  removeContact
}

export default PeopleService

File: api/src/main/kotlin/br/com/leuras/service/PeopleService.kt
package br.com.leuras.service

import br.com.leuras.exception.ModelNotFoundException
import br.com.leuras.model.User

interface PeopleService {
    fun search(keyword: String) : List<User>
    @Throws(ModelNotFoundException::class)
    fun contactsOf(user: User) : List<User>
    fun addContact(user: User, contact: User)
    fun removeContact(user: User, contact: User)
}